frontend/app-development/features/processEditor/ProcessEditor.tsx

```diff
@@ -81,13 +92,25 @@ export const ProcessEditor = (): React.ReactElement => {
     );
   };
 
+  const updateDataType = async (metaData: MetaDataForm): Promise<void> => {
+    mutateDataType(
+      { form: metaData },
+      {
+        onError: () => {
+          toast.error(t('process_editor.update_data_type_error'));
+        },
```

Comment:
Is it possible to create a test that checks that the error message is on the screen ? 😄

Reply:
I tried, but did not find a way to test it, unfortunately. Do you have any ideas?

Reply:
I'm not 100% sure, but after some research, I believe that if we create a test that calls on the ```updateDataType``` function, we could do something like this? 🤔 

```tsx
await waitFor(() => {
   const toastElement = getByRole('alert');
   expect(toastElement).toBeInTheDocument();
   expect(toastElement).toHaveTextContent(textMock('process_editor.update_data_type_error'));
});
```

Using ```getByRole('alert', { name: ... } )``` won't work because the option ```name``` is used to match against accessible name of the element, and since ```react-toastify``` does not use ARIA attributes to set the accessible name of the toast, the ```name``` option will not be useful 😄 

It might also be possible to mock the toast.error something like this: 

```tsx
jest.mock('react-toastify', () => ({
  toast: {
    error: jest.fn(),
  },
}))
```

I'm curious to see if any of the two works! 😄

Reply:
We can mock a failing api-response but I dont know how to simulate that the call has been made, in order to check that the correct text is in the document (or the alert element).
But is it necessary? 🫣
Wont we only test the functionality given by the external toast package?

Reply:
We can mock a failing api-response but I dont know how to simulate that the call has been made, in order to check that the correct text is in the document (or the alert element). I believe there is needed a lot of mock data in order to be able to render all the underlying components to be able to find the data model selector and trigger the updateDataType call 🤔 Especially since this component is a wrapper for the process-editor package
But is it necessary? 🫣
Wont we only test the functionality given by the external toast package?

Reply:
> We can mock a failing api-response but I dont know how to simulate that the call has been made, in order to check that the correct text is in the document (or the alert element). I believe there is needed a lot of mock data in order to be able to render all the underlying components to be able to find the data model selector and trigger the updateDataType call 🤔 Especially since this component is a wrapper for the process-editor package But is it necessary? 🫣 Wont we only test the functionality given by the external toast package?

I agree with you, was just curious to see how and if it was possible 😅 😄 

> Actually, I figured out that it is better to dispatch the data type changes from the metadata and simply pass it is since there is another endpoint that handles this change compared to the task id changes that must be wrapped inside a metadata object since it is passed together with the bpmn file.
So that means that we actually dont need this code here. And since we do not alter the bpmn file the error handling implemented in the sync utils will be sufficient in this case :D

That's great to hear! 🚀  Well done 👏